### ISSUE TITLE
Support generating objects from XSD files in quarkus-jaxb extension

### ISSUE THREAD

**Issue by rokkolesa** [quarkusio/quarkus#35207]
### Description

It would be very useful if the `quarkus-jaxb` extension would support generating Java objects from XSD files.
You could leverage the `xjc` tool which the `quarkus-cxf` extension already uses for `wsdl2java` generation.
It would also be useful to support `xjb` files as well to customize the object generation, or just in general to support additional params.

Right now, we have to use e.g. `org.codehaus.mojo:jaxb2-maven-plugin` to achieve this with the `xjc` goal.

(Moved here from quarkiverse/quarkus-cxf#951 according to @ppalaga ).

**Comment by wwwaldemar** [comment]
i am also interested on this feature ::)

**Comment by Bardioc1977** [comment]
Hello,

this would be awesome, so we no longer need a maven plugin to generate classes from XSD and can configure everything in the application.properties.

So I'd be interested as well.

**Comment by ppalaga** [comment]
Quarkus CXF actually supports generating Java model objects from XML Schema via `wsdl2java` that internally delegates to `xjc`. I'd need to check, whether we can generate only from XML Schema (without any WSDL involved) - I created an issue for that: https://github.com/quarkiverse/quarkus-cxf/issues/1842

We started testing the mixed scenario (WSDL includes an XSD file) since Quarkus CXF 3.23.0. 
Since that version, the generated classes can also be customized by `xjb` (bindings) files and xjc plugins from `org.jvnet.jaxb:jaxb-plugins` via [io.quarkiverse.cxf:quarkus-cxf-jaxb-plugins](https://docs.quarkiverse.io/quarkus-cxf/dev/reference/extensions/quarkus-cxf-jaxb-plugins.html).

@rokkolesa, @wwwaldemar, @Bardioc1977, @StFS, does anybody of you need just JAXB, without any SOAP functionality - so having plain `xcj` in `quarkus-jaxb` would still make sense for you?

**Comment by rokkolesa** [comment]
@ppalaga yes, one of the services I connect to actually has separate XSD files - outside of WSDL.

That is great news regarding the `quarkus-cxf-jaxb-plugins` as we are already using these plugins in our `org.jvnet.jaxb:jaxb-maven-plugin`!

**Comment by Bardioc1977** [comment]
> Quarkus CXF actually supports generating Java model objects from XML Schema via `wsdl2java` that internally delegates to `xjc`. I'd need to check, whether we can generate only from XML Schema (without any WSDL involved) - I created an issue for that: [quarkiverse/quarkus-cxf#1842](https://github.com/quarkiverse/quarkus-cxf/issues/1842)
> 
> We started testing the mixed scenario (WSDL includes an XSD file) since Quarkus CXF 3.23.0. Since that version, the generated classes can also be customized by `xjb` (bindings) files and xjc plugins from `org.jvnet.jaxb:jaxb-plugins` via [io.quarkiverse.cxf:quarkus-cxf-jaxb-plugins](https://docs.quarkiverse.io/quarkus-cxf/dev/reference/extensions/quarkus-cxf-jaxb-plugins.html).
> 
> [@rokkolesa](https://github.com/rokkolesa), [@wwwaldemar](https://github.com/wwwaldemar), [@Bardioc1977](https://github.com/Bardioc1977), [@StFS](https://github.com/StFS), does anybody of you need just JAXB, without any SOAP functionality - so having plain `xcj` in `quarkus-jaxb` would still make sense for you?

@ppalaga Yes, we have cases, where we only have an XSD or some XSDs for which we want to generate Java Classes, without any SOAP functionality. We utilise JAX-B to generate an XML which is afterwards send as Kafka Event or receive XML from Kafka as Payload.

**Comment by ppalaga** [comment]
I've just tried Quarkus CXF for the plain XSD to Java scenatio and it does not work as I explained in https://github.com/quarkiverse/quarkus-cxf/issues/1842#issuecomment-2965911255

So using `org.codehaus.mojo:jaxb2-maven-plugin` is the only way to do this atm.
